src/App.jsx

```diff
@@ -1,19 +1,26 @@
-import { useState } from "react";
-import reactLogo from "./assets/react.svg";
-import viteLogo from "/vite.svg";
+import { useContext, useState, useEffect } from "react";
 import "./App.css";
-import { Products } from "./components/Products";
 import {
-	Link,
-	Route,
-	Routes,
 	useNavigate,
-	BrowserRouter,
   Outlet,
 } from "react-router-dom";
+import { ThemeContext } from "./ThemeContext";
+import { ThemeProvider } from "./ThemeContext";
 
 function App() {
   const navigate = useNavigate();
+  
+  const [selectValue, setSelectValue] = useState("light")
```

Comment:
This extra `useState` was probably giving you some issues huh? You actually don't need it at all and we could also remove the `useEffect` on line 21. I have a feeling this is what you were striving for:

```
  const navigate = useNavigate();

  const { changeTheme, theme } = useContext(ThemeContext);

  useEffect(() => {
    document.body.style.backgroundColor = theme.background;
  }, [theme]);

  const handleShowHome = () => {
    navigate('home');
  };

  const handleShowProducts = () => {
    navigate('products');
  };

// You already have your option values set to match the theme name, so just set it directly using `changeTheme`
  const themeSelect = (event) => {
    const val = event.target.value;
    changeTheme(val);
  };

```